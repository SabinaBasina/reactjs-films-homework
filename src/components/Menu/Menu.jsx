import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Menu.scss';

const Menu = () => (
  <header className={styles.header}>

    <Link to="/" className={styles.tvShowsLink}>
      <button type="button" className={styles.tvShows}>
        TvShows
      </button>
    </Link>

    <div className={styles.auth}>
      <Link to="/login">
        <button type="button" className={styles.log}>
          Login
        </button>
      </Link>
    </div>

  </header>

);

export default Menu;
