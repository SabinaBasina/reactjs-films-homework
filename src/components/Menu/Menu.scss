$back-color: #222;

.header {
  background-color: $back-color;
  height: 55px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.tvShows,
.log,
.profile {
  outline: none;
  cursor: pointer;
  color: #9d9d9d;
  border: none;
  background-color: $back-color;
}

.log,
.profile {
  font-size: 18px;
  margin-top: 10px;
}

.auth {
  margin-right: 15%;
}

.tvShows {
  font-size: 35px;
  margin-top: 2px;
}

.tvShowsLink {
  margin-left: 15%;
}

.header button:hover {
  color: white;
}

@media only screen and (max-width: 820px) {
  .tvShowsLink {
    margin-left: 3%;
  }

  .auth {
    margin-right: 3%;
  }
}
