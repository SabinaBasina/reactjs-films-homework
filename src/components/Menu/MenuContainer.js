import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';

import Menu from './Menu';
import { getIsAuthentication } from '../../modules/moduleAuth/moduleAuthSelectors';

const mapStateToProps = createStructuredSelector({
  isAuthentication: getIsAuthentication,
});

export default connect(mapStateToProps)(Menu);
