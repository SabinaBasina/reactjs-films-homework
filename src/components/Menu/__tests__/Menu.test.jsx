import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Menu from '../Menu';

test('Menu snapsphot', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Menu />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
