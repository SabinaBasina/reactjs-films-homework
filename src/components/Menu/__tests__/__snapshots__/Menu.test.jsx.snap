// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapsphot 1`] = `
<Secure
  render={[Function]}
/>
`;
